### PR TITLE
hub/pgxhub: return the error in TestHubNode

### PR DESCRIPTION
**Pull request by dloveen-v**
Screenshot 2024-12-05 at 15 04 54

### CODE REVIEW

File: pkg/system/hub/pgxhub/server.go [hunk 1]
```diff
@@ -294,7 +294,7 @@ func (n *Server) TestHubNode(ctx context.Context, request *gen.TestHubNodeReques
 	_, err = client.GetMetadata(ctx, &traits.GetMetadataRequest{})
 	if err != nil {
 		logger.Debug("failed api request", zap.Error(err))
-		return nil, status.Error(codes.Unavailable, "failed api request")
+		return nil, status.Error(codes.Unavailable, fmt.Errorf("failed api request: %w", err).Error())
```
Comment (mattnathan) [inline pkg/system/hub/pgxhub/server.go:297]: I think you want `status.Errorf(codes.Unavailable, "failed api request: %v", err)`

**Review by mattnathan — CHANGES_REQUESTED** [review]

File: pkg/system/hub/pgxhub/server.go (hunk 1 above)
Comment (dloveen-v) [reply pkg/system/hub/pgxhub/server.go]: `%v` seemed to generic initially. The proper way to wrap errors is with `%w` but it only works in `fmt.Errorf` calls. I've amended.

Comment (mattnathan) [reply pkg/system/hub/pgxhub/server.go]: That's correct, but you aren't wrapping errors (as status.Error doesn't support that), you're formatting strings.

**Review by mattnathan — APPROVED** [review]
Can you make sure the changes are squashed before merging please

**Pull request merged**